src/support/string.cpp

```diff
@@ -432,13 +432,14 @@ bool isUTF8(std::string_view str) {
   return true;
 }
 
-std::vector<char> unescapeJSONToWTF8(const char* str) {
+std::vector<char> unescapeUTF8JSONtoWTF16(const char* str) {
```

Comment:
Other conversions return `bool` to report validity (and use replacement characters for invalid input) and write to an `ostream&`. It might be nice to be consistent with that pattern here. That would slightly simplify the creation of a `string_view` in the caller and would let you use `writeWTF16CodePoint` below rather than manually pushing a bunch of zero bytes.